### ISSUE TITLE
feat: support iOS simulator with `shorebird preview`

### ISSUE THREAD

**Issue by LeFrosch** [shorebirdtech/shorebird#1432]
As far as I can tell the `shorebird preview` command does not work with an iOS simulator. Since I do not one a physical iOS device this makes testing kinda hard.

**Comment by eseidel** [comment]
Correct.  Shorebird currently only supports Release builds of Flutter, which do not work on an iOS simulator.

It would be possible for us to build Flutter Release mode for an iOS simulator (despite the Flutter project's current choice not to), but we haven't set that up yet.

**Comment by btrautmann** [comment]
Ran into this today as well. If not already (I couldn't find it!), would be great to include this in the docs or even in the `shorebird preview` command, e.g after 10s of searching, print a message indicating that if the user is expecting a simulator to be found, that won't work.

**Comment by eseidel** [comment]
Fair.  I think this also just a missing feature from Flutter and we could pretty trivially add it (Release support for Simulators).  I know why we didn't add it originally (since simulators are not representative for performance), but I think it's mostly just confusing to users.

**Comment by btrautmann** [comment]
@eseidel Totally...As a user I wouldn't even mind you finger-pointing as it being a Flutter native limitation 😉 , but having some messaging or warning would've helped at least in my case 😄
